### PR TITLE
FIM System tests: Scenario 204, Whodata absent switch to realtime

### PR DESCRIPTION
**Pull request by Zenidd**
Hi team,

This PR fixes some bugs in the `204 Whodata ---> Realtime` scenario and also in common tasks:
- The `auditd` process is now uninstalled before the tests through a task
- A new task waits until whodata switchs to realtime. Another task waits until the initial frequency scan ends.
- For the scenario to work, as it has a large frequency value, the target `/opt/fim_testing` directory is created before the agent restarts so it can be monitored from the first scan.
- The target agent `family_os` is now checked to assign it the right (rpm|deb) `ossec.conf` file

Greetings, 
JP Sáez

### CODE REVIEW

File: tests/system/fim/common_tasks/prepare.yml [hunk 1]
```diff
@@ -39,6 +39,13 @@
     when:
       - inventory_hostname in groups['linuxagents']
 
+  - name: Create {{ files_fim_testing_linux }} to store outputs in Agents hosts | Linux
+    file:
+      path: "{{ files_fim_testing_linux }}"
+      state: directory
+    when:
+      - inventory_hostname in groups['linuxagents']
```
Comment (jm404) [inline tests/system/fim/common_tasks/prepare.yml]: Duplicate this task for Windows hosts please

File: tests/system/fim/common_tasks/prepare.yml [hunk 2]
```diff
@@ -50,16 +57,25 @@
...
       state: stopped
-
+      
   - name: Clean logs wazuh-agent
     become: True
     shell: "echo '' > {{ linux_ossec_folder_path }}/logs/ossec.log"
-    when: clean_logs
+    when: clean_logs      
+  
+  - name: Copy {{ files_config_agent_linux_ossec_conf_rpm }} to the Agents hosts | rpm | Linux
```
Comment (jm404) [inline tests/system/fim/common_tasks/prepare.yml]: Remove | Linux

File: tests/system/fim/scenarios/204_whodata_linux_noaudit/launch_test_scenario.yml [hunk 3]
```diff
@@ -8,45 +8,61 @@
...
-        - name: Check for disabled auditd. Check for realtime running instead.
-          become: true
-          import_tasks: "tests/tests.yml"
+        - name: Wait until whodata switch to realtime message.
+          become: True
+          wait_for:
+            path: "{{ linux_ossec_folder_path }}/logs/ossec.log"
+            search_regex: 'Who-data engine could not start. Switching who-data to real-time.'
+          when:
+            - inventory_hostname in groups['linuxagents']
```
Comment (jm404) [inline tests/system/fim/scenarios/204_whodata_linux_noaudit/launch_test_scenario.yml]: Please consider adding a custom timeout (default is 300s)

File: tests/system/fim/scenarios/204_whodata_linux_noaudit/launch_test_scenario.yml [hunk 4]
```diff
@@ -8,45 +8,61 @@
...
-        - block:
-          - name: Configure syscheck event.
-            set_fact:
-              event: "added"
+        - name: Wait until the first FIM scan end (Linux)
+          become: True
+          wait_for:
+            path: "{{ linux_ossec_folder_path }}/logs/ossec.log"
+            search_regex: 'Real-time file integrity monitoring started'
+          when:
```
Comment (jm404) [inline tests/system/fim/scenarios/204_whodata_linux_noaudit/launch_test_scenario.yml]: Please consider adding a custom timeout (default is 300s)

File: tests/system/fim/common_tasks/prepare.yml (hunk 1 above)
Comment (Zenidd) [reply tests/system/fim/common_tasks/prepare.yml]: The task has been duplicated for Windows hosts.

File: tests/system/fim/scenarios/204_whodata_linux_noaudit/launch_test_scenario.yml (hunk 3 above)
Comment (Zenidd) [reply tests/system/fim/scenarios/204_whodata_linux_noaudit/launch_test_scenario.yml]: The timeout is now 60 seconds

File: tests/system/fim/scenarios/204_whodata_linux_noaudit/launch_test_scenario.yml (hunk 4 above)
Comment (Zenidd) [reply tests/system/fim/scenarios/204_whodata_linux_noaudit/launch_test_scenario.yml]: The timeout is now 60 seconds

**Pull request merged**